pkg/compact/compact.go

```diff
@@ -249,67 +252,8 @@ func (s *Syncer) GarbageCollect(ctx context.Context) error {
 	return nil
 }
 
-func (s *Syncer) GarbageBlocks(resolution int64) (ids []ulid.ULID, err error) {
-	// Map each block to its highest priority parent. Initial blocks have themselves
-	// in their source section, i.e. are their own parent.
-	parents := map[ulid.ULID]ulid.ULID{}
-
-	for id, meta := range s.blocks {
-		// Skip any block that has a different resolution.
-		if meta.Thanos.Downsample.Resolution != resolution {
-			continue
-		}
-
-		// For each source block we contain, check whether we are the highest priority parent block.
-		for _, sid := range meta.Compaction.Sources {
-			pid, ok := parents[sid]
-			// No parents for the source block so far.
-			if !ok {
-				parents[sid] = id
-				continue
-			}
-			pmeta, ok := s.blocks[pid]
-			if !ok {
-				return nil, errors.Errorf("previous parent block %s not found", pid)
-			}
-			// The current block is the higher priority parent for the source if its
-			// compaction level is higher than that of the previously set parent.
-			// If compaction levels are equal, the more recent ULID wins.
-			//
-			// The ULID recency alone is not sufficient since races, e.g. induced
-			// by downtime of garbage collection, may re-compact blocks that are
-			// were already compacted into higher-level blocks multiple times.
-			level, plevel := meta.Compaction.Level, pmeta.Compaction.Level
-
-			if level > plevel || (level == plevel && id.Compare(pid) > 0) {
-				parents[sid] = id
-			}
-		}
-	}
-
-	// A block can safely be deleted if they are not the highest priority parent for
-	// any source block.
-	topParents := map[ulid.ULID]struct{}{}
-	for _, pid := range parents {
-		topParents[pid] = struct{}{}
-	}
-
-	for id, meta := range s.blocks {
-		// Skip any block that has a different resolution.
-		if meta.Thanos.Downsample.Resolution != resolution {
-			continue
-		}
-		if _, ok := topParents[id]; ok {
-			continue
-		}
-
-		ids = append(ids, id)
-	}
-	return ids, nil
-}
-
 func (s *Syncer) garbageCollect(ctx context.Context, resolution int64) error {
-	garbageIds, err := s.GarbageBlocks(resolution)
+	garbageIds, err := s.garbageBlocksFinder.GarbageBlocks(resolution, s.blocks)
```

Comment:
I believe this will do nothing as we have duplicate block filter already. We need to use that instead and remove GarbageCollect method. I thought that was the idea from the beginning (:

Reply:
removed the garbage filter